js/main.js

```diff
@@ -1,57 +1,55 @@
+chrome.storage.sync.get(null, function (items) {
+    var mySigns = (items.signs ? JSON.parse(items.signs) : []);
+    if (mySigns.length < 1) {
+        $("#exe").hide();
+        $("#lstSigns").hide();
+        $("#msg").html(chrome.i18n.getMessage("error_no_signs"));
+        document.getElementById("aConf").addEventListener("click", function () {
+            chrome.tabs.create({
+                url: "config.html"
+            });
+        });
+    } else {
+        for (var row in mySigns) {
+            var select = document.getElementById('lstSigns');
+            var opt = document.createElement('option');
+            opt.value = row;
+            opt.innerHTML = mySigns[row].name;
+            select.appendChild(opt);
+        }
+    }
 
+    document.getElementById("exe").addEventListener("click", function () {
+        chrome.tabs.query({
+            currentWindow: true,
+            active: true,
+            url: "https://productforums.google.com/*"
+        }, function (tab) {
+            if (tab.length < 1) {
+                alert(chrome.i18n.getMessage("error_no_gpf_tab"));
+                return;
+            }
 
-chrome.storage.sync.get(null, function(items) {
-  var mySigns = (items.signs ? JSON.parse(items.signs) : []);
-  if (mySigns.length < 1) {
-      $("#exe").hide();
-      $("#lstSigns").hide();
-      $("#msg").html(chrome.i18n.getMessage("error_no_signs"));
-      document.getElementById("aConf").addEventListener("click", function () {
-          chrome.tabs.create({
-              url: "config.html"
-          });
-      });
-  } else {
-      for (var row in mySigns) {
-          var select = document.getElementById('lstSigns');
-          var opt = document.createElement('option');
-          opt.value = row;
-          opt.innerHTML = mySigns[row].name;
-          select.appendChild(opt);
-      }
-  }
+            if (typeof mySigns[$("#lstSigns").children(":selected").attr("value")] == 'undefined') {
+                alert(chrome.i18n.getMessage("error_haha"));
+                return;
+            }
+            var sitem = mySigns[$("#lstSigns").children(":selected").attr("value")];
+            mySigns[$("#lstSigns").children(":selected").attr("value")].used++;
+            chrome.tabs.sendMessage(tab[0].id, {
+                type: "addSign",
+                tab: tab[0],
+                text: sitem.message,
+                "post": sitem.post
+            });
+            localStorage.setItem("signs", JSON.stringify(mySigns))
```

Comment:
La idea del localStorage es no tocar el límite de consultas del sync, aunque leyendo la documentación, MAX_WRITE_OPERATIONS_PER_HOUR, es solo para las acciones, *write, remove y clear*, por lo cual no existiría problema de implementar todo en la sync de Google. Sin embargo, @nsinisterra, mi comentario del localStorage es usarlo como caché local, no como contenedor definitivo. Si los cambios los escribes en el sync de Google y luego obtienes una copia de caché local, debería ir más rápido(?).

El límite de caracteres es un problema que [Firebase](https://firebase.google.com) podría ayudarnos a resolver (it's free!)